modules/search/class-jetpack-instant-search.php

```diff
@@ -0,0 +1,367 @@
+			<?php } ?>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Loads scripts for Tracks analytics library
+	 */
+	public function load_and_initialize_tracks() {
+		wp_enqueue_script( 'jp-tracks', '//stats.wp.com/w.js', array(), gmdate( 'YW' ), true );
+	}
+
+	/**
+	 * Get the version number to use when loading the file. Allows us to bypass cache when developing.
+	 *
+	 * @param string $file Path of the file we are looking for.
+	 * @return string $script_version Version number.
+	 */
+	public static function get_asset_version( $file ) {
+		return Jetpack::is_development_version() && file_exists( JETPACK__PLUGIN_DIR . $file )
+			? filemtime( JETPACK__PLUGIN_DIR . $file )
+			: JETPACK__VERSION;
+	}
+
+	/**
+	 * Bypass the normal Search query since we will run it with instant search.
+	 *
+	 * @since 8.3.0
+	 *
+	 * @param array    $posts Current array of posts (still pre-query).
+	 * @param WP_Query $query The WP_Query being filtered.
+	 *
+	 * @return array Array of matching posts.
+	 */
+	public function filter__posts_pre_query( $posts, $query ) {
+		if ( ! $this->should_handle_query( $query ) ) {
+			// Intentionally not adding the 'jetpack_search_abort' action since this should fire for every request except for search.
+			return $posts;
+		}
+
+		/**
+		 * Bypass the main query and return dummy data
+		 *  WP Core doesn't call the set_found_posts and its filters when filtering
+		 *  posts_pre_query like we do, so need to do these manually.
+		 */
+		$query->found_posts   = 1;
+		$query->max_num_pages = 1;
+
+		return array();
+	}
+
+	/**
+	 * Run the aggregations API query for any filtering
+	 *
+	 * @since 8.3.0
+	 *
+	 * @param WP_Query $query The WP_Query being filtered.
+	 */
+	public function action__parse_query( $query ) {
+		if ( ! empty( $this->search_result ) ) {
+			return;
+		}
+
+		if ( is_admin() ) {
+			return;
+		}
+
+		if ( empty( $this->aggregations ) ) {
+			return;
+		}
+
+		jetpack_require_lib( 'jetpack-wpes-query-builder/jetpack-wpes-query-builder' );
+
+		$builder = new Jetpack_WPES_Query_Builder();
+		$this->add_aggregations_to_es_query_builder( $this->aggregations, $builder );
+		$this->search_result = $this->instant_api(
+			array(
+				'aggregations' => $builder->build_aggregation(),
+				'size'         => 0,
+				'from'         => 0,
+			)
+		);
+	}
+
+	/**
+	 * Run an instant search on the WordPress.com public API.
+	 *
+	 * @since 8.3.0
+	 *
+	 * @param array $args Args conforming to the WP.com v1.3/sites/<blog_id>/search endpoint.
+	 *
+	 * @return object|WP_Error The response from the public API, or a WP_Error.
+	 */
+	public function instant_api( array $args ) {
+		$start_time = microtime( true );
+
+		// Cache locally to avoid remote request slowing the page.
+		$transient_name = '_jetpack_instant_search_cache_' . md5( wp_json_encode( $args ) );
+		$cache          = get_transient( $transient_name );
+		if ( false !== $cache ) {
+			$end_time = microtime( true );
```

Comment:
We are early returning so $end_time is not used? would it make sense to remove this line?

Reply:
cc @gibrown

Reply:
removed